pkg/store/image/image_test.go

```diff
@@ -71,20 +73,39 @@ func TestImageStore(t *testing.T) {
 	}
 
 	t.Logf("should be able to get image")
-	genTruncIndex := func(normalName string) string { return normalName[:(len(normalName)+1)/2] }
-	for id, img := range images {
-		got, err := s.Get(genTruncIndex(id))
-		assert.NoError(err)
-		assert.Equal(img, got)
+	for _, v := range images {
+		truncID := genTruncIndex(v.ID)
+		got, err := s.Get(truncID)
+		assert.NoError(err, "truncID:%s, fullID:%s", truncID, v.ID)
+		assert.Equal(v, got)
+	}
+
+	t.Logf("should be able to get image by truncated imageId without algorithm")
+	for _, v := range images {
+		truncID := genTruncIndex(v.ID[strings.Index(v.ID, ":")+1:])
+		got, err := s.Get(truncID)
+		assert.NoError(err, "truncID:%s, fullID:%s", truncID, v.ID)
+		assert.Equal(v, got)
+	}
+
+	t.Logf("should not be able to get image by ambiguous prefix")
+	ambiguousPrefix := "sha256"
+	for i := range images {
+		if i == 0 {
+			// skip the first item
```

Comment:
what's up here skipping the first item?

Reply:
Hmm. It is unnecessary, Maybe I is short-circuted on that time.:)
On that time I think the first item is only one image and it cann't be conflict with the other.
I will modify it.

Reply:
done